### PR TITLE
Docs updates

### PR DESCRIPTION
**Pull request by khewonc**
### What does this PR do?

Applying some of the docs suggestions from https://github.com/DataDog/datadog-operator/pull/801. Most of the suggestions can't be applied since the `.md` files are mostly autogenerated and the field descriptions are from packages we don't own.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: n/a
* Cluster Agent: n/a

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### Checklist

- [x] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [x] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

**Review by jhgilbert — APPROVED** [review]

File: apis/datadoghq/common/v1/agent_types.go [hunk 1]
```diff
@@ -19,9 +19,9 @@ type AgentImageConfig struct {
 	// Use "gcr.io/datadoghq/cluster-agent:latest" for Datadog Cluster Agent.
 	// Use "agent" with the registry and tag configurations for <registry>/agent:<tag>.
 	// Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag>.
-	// If the name is full image string - `<name>:<tag>` or `<registry>/<name>:<tag>`, `tag` and `jmxEnabled` values are ignored,
-	// so is `global.registry`.
-	// Otherwise, image string is created by overriding default settings with supplied `name`, `tag` and `jmxEnabled` values;
+	// If the name is full image string—`<name>:<tag>` or `<registry>/<name>:<tag>`, then `tag`, `jmxEnabled`,
```
Comment (celenechang) [inline apis/datadoghq/common/v1/agent_types.go:22]:
```suggestion
	// If the name is the full image string—`<name>:<tag>` or `<registry>/<name>:<tag>`, then `tag`, `jmxEnabled`,
```
there's a chance i'm not reading it correctly

File: hack/generate-docs/v2alpha1_overrides.markdown [hunk 2]
```diff
@@ -2,9 +2,9 @@
 
 ### Override
 
-Below table lists parameters which can be used to override default or global settings. Maps and arrays have a type annotation in the table; properties which are configured as map values contain a `[key]` element which should be replaced by actual map key. `override` itself is a map with following possible keys `nodeAgent`, `clusterAgent` or `clusterChecksRunner`. Other keys can be added but it will not have any effect.
+The table below lists parameters that can be used to override default or global settings. Maps and arrays have a type annotation in the table; properties that are configured as map values contain a `[key]` element which should be replaced by actual map key. `override` itself is a map with the following possible keys: `nodeAgent`, `clusterAgent`, or `clusterChecksRunner`. Other keys can be added, but they do not have any effect.
```
Comment (celenechang) [inline hack/generate-docs/v2alpha1_overrides.markdown:5]:
```suggestion
The table below lists parameters that can be used to override default or global settings. Maps and arrays have a type annotation in the table; properties that are configured as map values contain a `[key]` element which should be replaced by the actual map key. `override` itself is a map with the following possible keys: `nodeAgent`, `clusterAgent`, or `clusterChecksRunner`. Other keys can be added, but they do not have any effect.
```

File: hack/generate-docs/v2alpha1_overrides.markdown [hunk 3]
```diff
@@ -2,9 +2,9 @@
...
 
-For example below manifest can be used to override node agent image and tag and resource limits of the system probe container. 
+For example, the manifest below can be used to override node Agent image, tag, and the resource limits of the system probe container.
```
Comment (celenechang) [inline hack/generate-docs/v2alpha1_overrides.markdown:7]:
```suggestion
For example, the manifest below can be used to override the node Agent image, tag, and resource limits of the system probe container. 
```

File: hack/generate-docs/v2alpha1_overrides.markdown [hunk 4]
```diff
@@ -24,5 +24,5 @@ spec:
               cpu: "2"
               memory: 1Gi
 ```
-In the table, `spec.override.nodeAgent.image.name` and `spec.override.nodeAgent.containers.system-probe.resources.limits` will appear as `[key].image.name`, `[key].containers.[key].resources.limits` respectively.
+In the table, `spec.override.nodeAgent.image.name` and `spec.override.nodeAgent.containers.system-probe.resources.limits` appear as `[key].image.name` and `[key].containers.[key].resources.limits` respectively.
```
Comment (celenechang) [inline hack/generate-docs/v2alpha1_overrides.markdown:27]:
```suggestion
In the table, `spec.override.nodeAgent.image.name` and `spec.override.nodeAgent.containers.system-probe.resources.limits` appear as `[key].image.name` and `[key].containers.[key].resources.limits`, respectively.
```

**Review by celenechang — APPROVED** [review]

**Pull request merged**